src/main/kotlin/au/com/sealink/quickprint/ApplicationSecurity.kt

```diff
@@ -0,0 +1,24 @@
+package au.com.sealink.quickprint
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Configuration
+import org.springframework.core.env.Environment
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter
+
+@Configuration
+class ApplicationSecurity : WebSecurityConfigurerAdapter() {
+    @Autowired
+    private lateinit var env : Environment
+
+    override fun configure(http: HttpSecurity) {
+        val protectedRoutes = arrayOf("/print-tickets", "/printers")
+
+        http.csrf().disable()
+                .requestMatchers()
+                    .antMatchers(*protectedRoutes)
+                    .and()
+                .addFilterBefore(AuthenticationFilter(env.getProperty("secrets.apiKey") ?: ""), BasicAuthenticationFilter::class.java)
```

Comment:
Would we rather explode if we don't have a `secrets.apiKey` rather than just allow anyone to print with a blank `api_key` param??

Reply:
Interesting enough you can't start without the envar defined due to the way it is defined in the properties file, however I will make it more explicit here as well